rct229/rulesets/ashrae9012019/section11/section11rule12.py

```diff
@@ -0,0 +1,156 @@
+                rmd_b = data["rmd_b"]
+                is_leap_year_b = data["is_leap_year_b"]
+
+                bldg_open_sch_id_b = getattr_(
+                    building_b, "buildings", "building_open_schedule"
+                )
+
+                bldg_open_sch_b = find_exactly_one_schedule(rmd_b, bldg_open_sch_id_b)[
+                    "hourly_values"
+                ]
+
+                hours_this_year = (
+                    LeapYear.LEAP_YEAR_HOURS
+                    if is_leap_year_b
+                    else LeapYear.REGULAR_YEAR_HOURS
+                )
+
+                return {
+                    "bldg_open_sch_b": bldg_open_sch_b,
+                    "hours_this_year": hours_this_year,
+                }
+
+            def list_filter(self, context_item, data):
+                bldg_open_sch_b = data["bldg_open_sch_b"]
+                hours_this_year = data["hours_this_year"]
+
+                return sum(bldg_open_sch_b) == hours_this_year
+
+            class BuildingSegmentRule(RuleDefinitionListIndexedBase):
+                def __init__(self):
+                    super(
+                        Section11Rule12.RMDRule.BuildingRule.BuildingSegmentRule, self
+                    ).__init__(
+                        rmds_used=produce_ruleset_model_description(
+                            USER=False, BASELINE_0=True, PROPOSED=True
+                        ),
+                        each_rule=Section11Rule12.RMDRule.BuildingRule.BuildingSegmentRule.SWHRule(),
+                        index_rmd=BASELINE_0,
+                        list_path="$.zones[*].spaces[*].service_water_heating_uses[*]",
+                    )
+
+                def create_data(self, context, data):
+                    building_segment_p = context.PROPOSED
+                    rmd_p = data["rmd_p"]
+
+                    service_water_heating_use_ids_list_p = (
+                        get_swh_uses_associated_with_each_building_segment(
+                            rmd_p, building_segment_p["id"]
+                        )
+                    )
+
+                    return {
+                        "service_water_heating_use_ids_list_p": service_water_heating_use_ids_list_p
+                    }
+
+                def list_filter(self, context_item, data):
+                    swh_use_p = context_item.PROPOSED
+                    served_by_distribution_system_p = getattr_(
+                        swh_use_p,
+                        "service_water_heating_uses",
+                        "served_by_distribution_system",
+                    )
+                    service_water_heating_use_ids_list_p = data[
+                        "service_water_heating_use_ids_list_p"
+                    ]
+
+                    return (
+                        served_by_distribution_system_p
+                        in service_water_heating_use_ids_list_p
+                    )
+
+                class SWHRule(PartialRuleDefinition):
```

Comment:
I don't think the rule shall be evaluated at service water use level. Instead, I think it should be at the Building level.
if you read the applicable message, it starts with `the building` not `the water use`.

Also, if we check the logic - a great amount of logic is to verify whether the building has SWH load and it is operate 24 hours.

Reply:
Good point. Addressed.